client/src/app/pages/applications/application-assessment/components/application-assessment-wizard/application-assessment-wizard.tsx

```diff
@@ -184,76 +189,191 @@ export const ApplicationAssessmentWizard: React.FC<
   const onInvalid = (errors: FieldErrors<ApplicationAssessmentWizardValues>) =>
     console.error("form errors", errors);
 
-  const onSubmit = (formValues: ApplicationAssessmentWizardValues) => {
-    if (!assessment?.application?.id) {
-      console.log("An assessment must exist in order to save the form");
-      return;
+  const buildSectionsFromFormValues = (
+    formValues: ApplicationAssessmentWizardValues
+  ): Section[] => {
+    if (!formValues || !formValues[QUESTIONS_KEY]) {
+      return [];
     }
-
-    const saveAction = formValues[SAVE_ACTION_KEY];
-    const assessmentStatus: AssessmentStatus =
-      saveAction !== SAVE_ACTION_VALUE.SAVE_AS_DRAFT ? "COMPLETE" : "STARTED";
-
-    const payload: Assessment = {
-      ...assessment,
-      // stakeholders: formValues.stakeholders,
-      // stakeholderGroups: formValues.stakeholderGroups,
-
-      sections:
-        matchingQuestionnaire?.sections?.map((section) => {
-          // const commentValues = values["comments"];
-          // const fieldName = getCommentFieldName(category, false);
-          // const commentValue = commentValues[fieldName];
-          return {
-            ...section,
-            // comment: commentValue,
-            questions: section.questions.map((question) => ({
+    const updatedQuestionsData = formValues[QUESTIONS_KEY];
+
+    // Create an array of sections based on the questionsData
+    const sections: Section[] =
+      matchingQuestionnaire?.sections?.map((section) => {
+        // const commentValues = values["comments"];
+        // const fieldName = getCommentFieldName(category, false);
+        // const commentValue = commentValues[fieldName];
```

Comment:
Should we remove the commented out code here?

Reply:
I'm not sure what we are doing about comments since they are yet another thing not available on the new API. Wa just going to leave them for now.

Reply:
I see, ok. I wonder if we should leave a TODO then.